src/client/dfuse/dfuse_main.c

```diff
@@ -168,6 +168,11 @@ main(int argc, char **argv)
 		}
 	}
 
+	if (!dfuse_info->di_foreground && getenv("PMIX_RANK")) {
```

Comment:
we shouldn't have any dependency on PMIX or any MPI implementation / launcher.
we could be using slurm or pmi 1 or 2 for launchers, and we would bypass this check in this case anyway.

Reply:
This doesn't add a dependency, just stops it running in the background in cases where it wouldn't work.  If and when we add support for slurm or other launchers we could add an environment variable here but I thought that support was being removed anyway?

Reply:
sorry i didn't mean adding a dependency. i know checking an env variable doesn't add one.
my point is that you can't assume through this env that you should be running in foreground. there is nothing we need to do to support using other launchers like slurm. people can just use them and probably will run into the same issue when using orterun.

Reply:
If you can run "srun env" and show me a env that we could check in that case.

The point is that this PR changes the default behaviour (which is OK, in fact it's the intention) but it also breaks dfuse completely if it's run under PMIx which is the recommended way in the documentation currently so without this nobody is going to be able to run dfuse until they ask us for the new options to use.

Reply:
Actually, it would probably make more sense to check for the DAOS_SINGLETON_CLI env var not being set.

Reply:
i guess my point is that you just need to update the documentation with this patch, and not try to handle different launchers whether they work or no.
If you want to keep this check for PMIX since that's what we tell people to launch dfuse with in the doc now to keep the current behavior, then fine..  but my point is that most clients are not going to be using ompi/pmix and some other launcher that might not work without foreground mode. So the minimum you have to do here is to update the documentation..

Reply:
I agree that we need to update the documentation, both to advise on using singleton mode and to add this new flag but I was viewing that (or at least the first issue) as a separate item.

My concern is not so much that we differ from the documentation although that would be a problem but rather that current users found that their existing workflow no longer worked for non-obvious reasons.

Reply:
> Actually, it would probably make more sense to check for the DAOS_SINGLETON_CLI env var not being set.

that actually doesn't guarantee it will work since you can still use orterun with Singleton mode.. on the DAOS side, it won't use PMIX mode, but still will cause problem with dfuse without foreground mode.

Reply:
That's just compatibility code that will be removed when we move away from orterun.